### PR TITLE
test(): added some basic tests for rearrange and journal

### PR DESCRIPTION
**Pull request by xeladotbe**
This PR adds a few basic tests for the journal and the rearrange function.

### CODE REVIEW

**Comment by xeladotbe** [comment]
I have read the CLA Document and I hereby sign the CLA

**Comment by jamie-lemon** [comment]
@xeladotbe Looks good to me - however I will let @mipo1357 review as he did the initial commit for the testing code.

File: examples/tests/src/journal.test.ts [hunk 1]
```diff
@@ -0,0 +1,73 @@
+import * as fs from "fs";
+import * as mupdf from "mupdf";
+import path from "path";
+import { afterEach, beforeEach, describe, expect, it } from "vitest";
+
+const scriptdir = path.resolve(__dirname);
+const filename = path.join(scriptdir, "..", "test.pdf");
+const data = fs.readFileSync(filename);
+
+describe("mupdfjs metadata tests", () => {
```
Comment (mipo1357) [inline examples/tests/src/journal.test.ts:10]: Name it as a test case for journal operations.

File: examples/tests/src/rearrange.test.ts [hunk 2]
```diff
@@ -0,0 +1,41 @@
+import * as fs from "fs";
+import * as mupdf from "mupdf";
+import path from "path";
+import { afterEach, beforeEach, describe, expect, it } from "vitest";
+
+const scriptdir = path.resolve(__dirname);
+const filename = path.join(scriptdir, "..", "test.pdf");
+const data = fs.readFileSync(filename);
+
+describe("mupdfjs metadata tests", () => {
```
Comment (mipo1357) [inline examples/tests/src/rearrange.test.ts:10]: Name it as a test case for rerrange operations.

File: examples/tests/src/rearrange.test.ts [hunk 3]
```diff
@@ -0,0 +1,41 @@
...
+  });
+
+  it("should rearrange all pages in document", () => {
+    document.rearrangePages([2, 1, 0]);
+
+    expect(document.countPages()).toBe(3);
+    expect(document.loadPage(0)).toBeInstanceOf(mupdf.PDFPage);
+    expect(document.loadPage(1)).toBeInstanceOf(mupdf.PDFPage);
+    expect(document.loadPage(2)).toBeInstanceOf(mupdf.PDFPage);
+  });
```
Comment (mipo1357) [inline examples/tests/src/rearrange.test.ts:31]: What about checking against the content of the page to make sure the content has been replaced? What do you think? For example, some text, etc.

**Review by mipo1357 — COMMENTED** [review]
I appreciate your help very much.
Could you please check my review comments?

File: examples/tests/src/journal.test.ts (hunk 1 above)
Comment (xeladotbe) [reply examples/tests/src/journal.test.ts]: Sorry, I just copied your metadata file and forgot to adjust the name. I have corrected the name.

File: examples/tests/src/rearrange.test.ts (hunk 2 above)
Comment (xeladotbe) [reply examples/tests/src/rearrange.test.ts]: Sorry, I just copied your metadata file and forgot to adjust the name. I have corrected the name.

File: examples/tests/src/rearrange.test.ts (hunk 3 above)
Comment (xeladotbe) [reply examples/tests/src/rearrange.test.ts]: I am now comparing the JSON return value of toStructuredText, is that what you meant?

Comment (mipo1357) [reply examples/tests/src/rearrange.test.ts]: That's good.

**Pull request merged**